contrib/endpoints/src/api_manager/context/service_context.h

```diff
@@ -67,7 +67,10 @@ class ServiceContext {
 
   bool IsRulesCheckEnabled() const {
     return RequireAuth() && service().apis_size() > 0 &&
-           !config_->GetFirebaseServer().empty();
+        (!config_->GetFirebaseServer().empty() ||
```

Comment:
I agree with Wayne that the logic should be in config::GetFirebaseServer(). This function should be a wrapper, and determine whether the server is set from server config or service config.